fix: BROS-514: Data corruption during transformation

nick-skriabin opened the pull request:
This PR addresses the following two issues:

- Exception "Maximum update depth exceeded" when selecting a shape
- Data corruption during transformation due to shape deformation when bezier curves are used

<!--

This description MUST be filled out for a PR to receive a review. Its primary purposes are:

 - to enable your reviewer to review your code easily, and
 - to convince your reviewer that your code works as intended.

Some pointers to think about when filling out your PR description:
 - Reason for change: Description of problem and solution
 - Screenshots: All visible changes should include screenshots.
 - Rollout strategy: How will this code be rolled out? Feature flags / env var / other
 - Testing: Description of how this is being verified
 - Risks: Are there any known risks associated with this change, eg to security or performance?
 - Reviewer notes: Any info to help reviewers approve the PR
 - General notes: Any info to help onlookers understand the code, or callouts to significant portions.

You may use AI tools such as Copilot Actions to assist with writing your PR description (see https://docs.github.com/en/copilot/using-github-copilot/using-github-copilot-for-pull-requests/creating-a-pull-request-summary-with-github-copilot); however, an AI summary isn't enough by itself. You'll need to provide your reviewer with strong evidence that your code works as intended, which requires actually running the code and showing that it works.

-->

nick-skriabin committed:
Fix maximum update depth exceeded

nick-skriabin committed:
Fix data corruption during transformation

nick-skriabin committed:
Biome

nick-skriabin committed:
Merge branch 'develop' into fb-bros-514

nick-skriabin committed:
Remove unnecessary else

niklub commented:
/fm sync
> [Workflow run](https://github.com/HumanSignal/label-studio/actions/runs/18691249234)

niklub committed:
Sync Follow Merge dependencies

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/18691249234

niklub committed:
Merge branch 'develop' into 'fb-bros-514'

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/18691249234
